rpc/src/optimistically_confirmed_bank_tracker.rs

```diff
@@ -35,7 +35,34 @@ impl OptimisticallyConfirmedBank {
 pub enum BankNotification {
     OptimisticallyConfirmed(Slot),
     Frozen(Arc<Bank>),
-    Root(Arc<Bank>),
+    /// Send the Bank currently rooted and optionally the parent slots in (Slot, Parent Slot) pairs
+    /// The parent slots contains the slot of the bank.
+    Root((Arc<Bank>, Option<Vec<(Slot, Slot)>>)),
+}
+
+#[derive(Clone, Debug)]
+pub enum SlotNotification {
+    OptimisticallyConfirmed(Slot),
+    /// The (Slot, Parent Slot) pair for the slot frozen
+    Frozen((Slot, Slot)),
+    /// The (Slot, Parent Slot) pair for the root slot
+    Root((Slot, Slot)),
+}
+
+impl From<BankNotification> for SlotNotification {
+    fn from(notification: BankNotification) -> Self {
+        match notification {
+            BankNotification::OptimisticallyConfirmed(slot) => {
+                SlotNotification::OptimisticallyConfirmed(slot)
+            }
+            BankNotification::Frozen(bank) => {
+                SlotNotification::Frozen((bank.slot(), bank.parent_slot()))
+            }
+            BankNotification::Root(bank) => {
+                SlotNotification::Root((bank.0.slot(), bank.0.parent_slot()))
+            }
+        }
+    }
```

Comment:
Is this implementation necessary? Seem like it's not being used anywhere

Reply:
Removed